### PR TITLE
Renovate settings for replace directive deps and cilium/cilium updates

### PR DESCRIPTION
**Pull request by mtardy**
Fix https://github.com/cilium/tetragon/issues/1326.

### CODE REVIEW

File: .github/renovate.json5 [hunk 1]
```diff
@@ -295,6 +295,53 @@
...
+      "matchPackageNames": [
+        "github.com/cilium/cilium",
+      ],
+      "matchUpdateTypes": [
+        "major",
+        "minor",
+      ],
+      "draftPR": true,
+      "prBodyNotes": [
+        ":warning: Manual steps need to be performed (explaining why this PR is only a draft). First replace directives must be synchronized with cilium/cilium, see the `go.mod` and `pkg/k8s/go.mod` for more info. Then dependencies ignore in `.github/renovate.json5` for the replace directive must be updated accordingly."
```
Comment (lambdanis) [inline .github/renovate.json5:331]: IIUC it also requires updating k8s dependencies. Should renovate mention that in the draft PR too?

File: .github/renovate.json5 [hunk 2]
```diff
@@ -295,6 +295,53 @@
...
+      // do not update dependencies from the replace directive from cilium/cilium
+      // see go.mod second replace directive comments for more info
+      "enabled": false,
+      "matchPackageNames": [
+        // All of these packages are maintained on a Cilium fork. Thus, we don't
+        // want to update them automatically.
+        "go.universe.tf/metallb",
+        "github.com/cilium/metallb",
+        "github.com/vishvananda/netlink",
+        "github.com/cilium/netlink",
```
Comment (lambdanis) [inline .github/renovate.json5:309]: These two can be removed, as they are not replaced anymore (https://github.com/cilium/cilium/pull/27342 removes them from Cilium too).

**Review by lambdanis — APPROVED** [review]
Looks good, I left some minor comments, but approving anyway.

File: .github/renovate.json5 (hunk 1 above)
Comment (mtardy) [reply .github/renovate.json5:331]: So this should be automatic if you go get cilium or if you bump and go mod tidy. But maybe I should mention it indeed.

**Pull request merged**